### DISCUSSION TITLE
Get Id of Table Modal Action

### DISCUSSION THREAD

**Discussion by lutfifirdaus** [filamentphp/filament#5440] [Help] [answered]
Hello, Is anybody know how to get Id of modal action so I can emit event to close it on my custom function from other class? Or is it imposible to do that?

**Comment by jensjns** [comment]
Did you solve this?
Also looking for an answer to this

**Reply by lutfifirdaus** [reply, marked as answer]
I'm using modal blade component from filament core concept as my modal, so u can set you're Id by yourself, check in [here](https://filamentphp.com/docs/3.x/support/blade-components/modal)

**Reply by jensjns** [reply]
Thanks for the quick reply! I'm triggering the modal programmatically via a `Tables\Actions\Action `, seems like a custom ID isn't possible but I'll keep looking.

**Reply by lutfifirdaus** [reply]
Your welcome.

**Reply by bimasaktikr** [reply]
can we trigger modal `use  Tables\Actions\Action` ?

**Reply by kayleeherrmann** [reply]
> Thanks for the quick reply! I'm triggering the modal programmatically via a `Tables\Actions\Action `, seems like a custom ID isn't possible but I'll keep looking.

@jensjns Did you find a solution for this? 

I have re asked the question [here](https://github.com/filamentphp/filament/discussions/15239) since its slightly different to this thread (not using the blade component)  - however I thought I would reach out just in case.

**Comment by boris-glumpler** [comment]
I just needed this, so here's a way to get the id for an action modal, set it on the modal content, so you can then close it from within `some.view`:

```php
Action::make('media')
    // ... other action config
    ->modalContent(fn (Action $action, Page $livewire) => view('some.view', [
        'modalId' => "fi-{$livewire->getId()}-action-{$action->getNestingIndex()}",
    ]));
```

Works in v4.

**Comment by thekassdag** [comment]
You can get the **modal ID** of any Filament  Action inside your `ManageRecords` class using:

```php
$modalId = "fi-{$this->getId()}-action-{$this->getOriginallyMountedActionIndex()}";
```

Then you can **dispatch a close event** for that modal:

```php
$this->dispatch('close-modal', id: $modalId);
```


---

### **Example 1 — Close modal from `->action()`**

```php
Action::make('example')
    // ... other action config
    ->action(function () {
        // Get ID of the modal for this action
        $modalId = "fi-{$this->getId()}-action-{$this->getOriginallyMountedActionIndex()}";

        // Close the modal programmatically
        $this->dispatch('close-modal', id: $modalId);
    });
```

**Use case:** close the modal immediately when the action is triggered (e.g., a button click).

---

### **Example 2 — Close modal on Select field change**

```php
Action::make('example')
    // ... other action config
    ->modalHeading('Example Modal')
    ->modalContent([
        Select::make('some_field')
            ->label('Choose something')
            ->reactive()
            ->afterStateUpdated(function ($state) {
                // Get ID of the modal for this action
                $modalId = "fi-{$this->getId()}-action-{$this->getOriginallyMountedActionIndex()}";

                // Close the modal programmatically
                $this->dispatch('close-modal', id: $modalId);
            })
            ->searchable(),
    ]);
```

**Use case:** automatically close the modal when the user selects a value from a dropdown.

---
